### PR TITLE
Automatically create binary releases

### PR DESCRIPTION
**Pull request by cmichi**
Closes https://github.com/paritytech/canvas-node/issues/25.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -58,14 +59,34 @@ workflow:
...
+    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/        # i.e. v1.0, v2.1rc1
+    - if: $CI_COMMIT_REF_NAME == "master"
+      when: manual
+      allow_failure: true
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                   # PRs
+      when: manual
+      allow_failure: true
   script:
     - cargo build --release
-    - cargo test --release --all
```
Comment (TriplEight) [inline .gitlab-ci.yml:66]: what was wrong with keeping a test here? You don't even add additional flags, so it's perfect to build and test in the same job while reusing the cache.
And the mac build you can place after this job.
And publish job would only go after mac job and publish everything if the mac job is a success.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -58,14 +59,34 @@ workflow:
...
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "web"
+    - if: $CI_PIPELINE_SOURCE == "schedule"
+    - if: $CI_COMMIT_REF_NAME == "tags"
+    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/        # i.e. v1.0, v2.1rc1
+    - if: $CI_COMMIT_REF_NAME == "master"
+      when: manual
+      allow_failure: true
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                   # PRs
+      when: manual
```
Comment (TriplEight) [inline .gitlab-ci.yml:81]: this is why your build jobs are waitnig for the manual action

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -56,32 +49,98 @@ workflow:
...
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                         # PRs
+    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/              # i.e. v1.0, v2.1rc1
+
+.publish-refs:                     &publish-refs
+  rules:
+    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/              # i.e. v1.0, v2.1rc1
 
-build-test:
-  stage:                           test
+### stage:                         build
```
Comment (TriplEight) [inline .gitlab-ci.yml:76]:
```suggestion
### stage:                         build-linux
```

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -81,8 +81,8 @@ build-linux:
   <<:                              *collect-artifacts
   <<:                              *build-refs
   script:
-    - time cargo build --release
-    - time cargo test --release --all
```
Comment (TriplEight) [inline .gitlab-ci.yml]: what was wrong with it?

**Review by TriplEight — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 4 above)
Comment (cmichi) [reply .gitlab-ci.yml]: I thought this was what you meant by `just remove the debug lines when you're done` :-) ?

Comment (TriplEight) [reply .gitlab-ci.yml]: it's mostly needed for `cargo test` because no one knows how much the tests take up.

**Review by TriplEight — APPROVED** [review]

**Pull request merged**